### PR TITLE
Bump to 2.4.2

### PR DESCRIPTION
**Pull request by jezdez**
frozendict 2.4.2

**Destination channel:** defaults

### Links

- [PKG-4404](https://anaconda.atlassian.net/browse/PKG-4404)
- [Upstream repository](https://github.com/Marco-Sulla/python-frozendict)
- [Upstream changelog/diff](https://github.com/Marco-Sulla/python-frozendict/compare/v2.3.4...v2.4.2)
- Relevant dependency PRs:
  - https://github.com/conda/conda/pull/13766

### Explanation of changes:

- frozendict 2.3.5 introduces a C extension which vastly speeds up conda

[PKG-4404]: https://anaconda.atlassian.net/browse/PKG-4404?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by JeanChristopheMorinPerso** [comment]
The CI is failing. I'll take a look.

**Comment by JeanChristopheMorinPerso** [comment]
@jezdez It looks like they only have a C extension for 3.8, 3.9 and 3.10 and they only ship wheels python < 3.11. I'll edit the feedstock accordingly.

**Comment by skupr-anaconda** [comment]
> @jezdez It looks like they only have a C extension for 3.8, 3.9 and 3.10 and they only ship wheels python < 3.11. I'll edit the feedstock accordingly.

Yeah, there is an upstream issue https://github.com/Marco-Sulla/python-frozendict/issues/68

if you want the pure py implementation:
`FROZENDICT_PURE_PY=1 pip install frozendict`

see https://github.com/Marco-Sulla/python-frozendict/blob/a63cad5970da37f45ab9b270b6ac91478c7656d9/README.md#install

**Comment by skupr-anaconda** [comment]
I wonder how `conda-forge` could build it even if they got the same [error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=915203&view=logs&j=517fe804-fa30-5dc2-1413-330699242c05&t=c7d0fcb4-3a10-5b26-dfea-e879a93fc2ab&l=707) but without failure.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,47 +1,58 @@
...
+{% set sha256 = "741779e1d1a2e6bb2c623f78423bd5d14aad35dc0c57e6ccc89e54eaab5f1b8a" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:11]: I think `pypi.io` is a preferred domain in the `aggregate` repo:
```
╰─❯ rg "url: https://pypi.io" | wc
    1390   12952  183252
╰─❯ rg "url: https://pypi.org" | wc
      90     730   10747
```
Were there any troubles with `pypi.io`?

**Review by skupr-anaconda — APPROVED** [review]
LGTM but `source/url`

**Review by JeanChristopheMorinPerso — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (jezdez) [reply recipe/meta.yaml:11]: pypi.io has been deprecated for years, pypi.org is the canonical one. https://github.com/pypi/warehouse/issues/2184#issuecomment-315175958

**Comment by jezdez** [comment]
Thank you @JeanChristopheMorinPerso @skupr-anaconda 💚

**Pull request merged**